components/camel-azure/camel-azure-storage-blob/src/main/java/org/apache/camel/component/azure/storage/blob/BlobUtils.java

```diff
@@ -33,11 +33,26 @@ public static Message getInMessage(final Exchange exchange) {
         return ObjectHelper.isEmpty(exchange) ? null : exchange.getIn();
     }
 
-    public static Long getInputStreamLength(final InputStream inputStream) throws IOException {
-        final long length = IOUtils.toByteArray(inputStream).length;
-        inputStream.reset();
-
-        return length;
+    public static long getInputStreamLength(InputStream is) throws IOException {
+        if (!is.markSupported()) {
+            return -1;
```

Comment:
I'm afraid that returning -1 as length could cause side effects in some parts of the code